### PR TITLE
(wip) Add rustfmt.toml

### LINKED ISSUE #47: Add rustfmt.toml
> I support using rustfmt (with/without a `rustfmt.toml`) if you decide you want this. makes things easy.

_Originally posted by @pickx in https://github.com/hecatia-elegua/bilge/pull/44#discussion_r1257252277_

Adding rustfmt.toml and then adjusting the rules until it mostly doesn't want to format any of the current code would be a good way to do this.
There should be some option to say "line length can be huge" or even "I don't care about line length at all".

### PR DESCRIPTION
**Pull request by hecatia-elegua**
Resolves #47.
I started by adjusting the config file for some time, then ran `cargo fmt` once.
I've reverted some of the "chain"-ges until I gave up, but you can mostly see which parts are dumb.

### CODE REVIEW

File: bilge-impl/src/bitsize_internal.rs [hunk 1]
```diff
@@ -44,24 +45,24 @@ fn generate_struct(struct_data: &ItemStruct, arb_int: &TokenStream) -> TokenStre
...
+    let (accessors, (constructor_args, constructor_parts)): (Vec<TokenStream>, (Vec<TokenStream>, Vec<TokenStream>)) = fields
+        .iter()
         .map(|field| {
             // offset is needed for bit-shifting
             // struct Example { field1: u8, field2: u4, field3: u4 }
             // previous_field_sizes = []     -> unwrap_or_else -> field_offset = 0
             // previous_field_sizes = [8]    -> reduce         -> field_offset = 0 + 8     =  8
             // previous_field_sizes = [8, 4] -> reduce         -> field_offset = 0 + 8 + 4 = 12
-            let field_offset = previous_field_sizes.iter().cloned().reduce(|acc, next| quote!(#acc + #next)).unwrap_or_else(|| quote!(0));
+            let field_offset = previous_field_sizes.iter().cloned()
```
Comment (hecatia-elegua) [inline bilge-impl/src/bitsize_internal.rs:56]: e.g. I've reverted the formatting here, I would like it to fill the width I've set and only then wrap to a new line.
I think right now you can only tell rustfmt to wrap all chained calls if the width is overflown.

File: bilge-impl/src/bitsize_internal.rs [hunk 2]
```diff
@@ -101,11 +102,14 @@ fn generate_field(field: &Field, field_offset: &TokenStream, fieldless_next_int:
...
-        return (quote!(#getter), (quote!(), quote! { {
+        let accessors = quote!(#getter);
+        let constructor_arg = quote!();
+        let constructor_part = quote! { {
             // we still need to shift by the element's size
             offset += #size;
             0
-        } }))
+        } };
+        return (accessors, (constructor_arg, constructor_part));
```
Comment (hecatia-elegua) [inline bilge-impl/src/bitsize_internal.rs:112]: had to do this so it doesn't look weird, but I think it's better this way even.

File: bilge-impl/src/bitsize_internal/struct_gen.rs [hunk 3]
```diff
@@ -61,13 +61,15 @@ pub(crate) fn generate_getter_value(ty: &Type, offset: &TokenStream, is_array_el
...
+///
 /// `is_getter` allows us to generate a try_from impl more easily
 pub(crate) fn generate_getter_inner(ty: &Type, is_getter: bool) -> TokenStream {
     use Type::*;
     match ty {
         Tuple(tuple) => {
-            let unbraced = tuple.elems.iter()
+            let unbraced = tuple
+                .elems
+                .iter()
```
Comment (hecatia-elegua) [inline bilge-impl/src/bitsize_internal/struct_gen.rs:72]: ^this is dumb

File: examples/basic_enums.rs [hunk 4]
```diff
@@ -6,13 +6,18 @@ use bilge::prelude::*;
...
 }
 
 #[bitsize(2)]
 #[derive(FromBits)]
 enum CompleteEnum {
-    A, B, C, D
+    A,
+    B,
+    C,
+    D,
```
Comment (hecatia-elegua) [inline examples/basic_enums.rs:20]: ^this is dumb

File: src/lib.rs [hunk 5]
```diff
@@ -3,18 +3,24 @@
...
+        assume_filled,
+        bitsize,
+        bitsize_internal,
+        BinaryBits,
+        Bitsized,
+        DebugBits,
+        DefaultBits,
+        Filled,
+        FromBits,
+        TryFromBits,
```
Comment (hecatia-elegua) [inline src/lib.rs:23]: ^this is kinda dumb

Comment (pickx) [reply src/lib.rs:23]:
```rust
#[doc(no_inline)]
pub use super::{bitsize, bitsize_internal, Bitsized};
pub use super::{DebugBits, FromBits, TryFromBits, BinaryBits};
pub use super::arbitrary_int::*;
pub use super::{Filled, assume_filled};
```
this seems reasonable (except rustfmt decides to be jerk and reorders them, oh well)

Comment (hecatia-elegua) [reply src/lib.rs:23]: Though they all need no_inline I think. Actually I haven't looked at doc output in a while, we should do that.

File: bilge-impl/src/bitsize_internal.rs (hunk 1 above)
Comment (hecatia-elegua) [reply bilge-impl/src/bitsize_internal.rs:56]: see https://github.com/rust-lang/rustfmt/issues/2263#issuecomment-1648480213

**Comment by hecatia-elegua** [comment]
will squash rename the wip commits after review

**Review by pickx — COMMENTED** [review]
whew. looks good. 👍

**Review by pickx — APPROVED** [review]

**Pull request merged**